libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/back_insert_iterator.h

```diff
@@ -35,31 +35,39 @@ _LIBCUDACXX_SUPPRESS_DEPRECATED_PUSH
 template <class _Container>
 class _LIBCUDACXX_TEMPLATE_VIS back_insert_iterator
 #if _LIBCUDACXX_STD_VER <= 14 || !defined(_LIBCUDACXX_ABI_NO_ITERATOR_BASES)
-    : public iterator<output_iterator_tag,
-                      void,
-                      void,
-                      void,
-                      void>
+    : public iterator<output_iterator_tag, void, void, void, void>
 #endif
 {
 _LIBCUDACXX_SUPPRESS_DEPRECATED_POP
 protected:
     _Container* container;
```

Comment:
question: in the standard, this seems to be `_Container* container = nullptr;`. Having it initialized might help diagnoze issues. Is there anything preventing us from initializing pointer?

I'm also unable to find default constructor:
```cpp
constexpr back_insert_iterator() noexcept = default;
```
Should there be one? Same questions apply to remaining iterators.

Reply:
So we intentionally do not provide a default constructor exactly for the reason that we always want `container` to be properly initialized: https://en.cppreference.com/w/cpp/iterator/back_insert_iterator/back_insert_iterator

Given that this is essentially dead code from our libcu++ fork I would rather keep it as is and close to upstream

Reply:
Oh, it seems that I was looking at older draft, thank you for the reference!